staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/validation.go

```diff
@@ -75,6 +76,46 @@ func ValidateStructural(s *Structural, fldPath *field.Path) field.ErrorList {
 	return allErrs
 }
 
+// ValidateStructuralUpdate checks invariants of a structural schema on update or creation.
+// in publishing, we make old being nil to keep backwards compatibility.
+// TODO: reuse the function in v1 validation
+func ValidateStructuralUpdate(s *Structural, old *Structural, fldPath *field.Path) field.ErrorList {
```

Comment:
I don't think this needs both the old and new object. Something simpler like checking if the schema has invalid types and avoiding serving openapi could be done with a method like this (included in https://github.com/kubernetes/kubernetes/pull/79604):

https://github.com/liggitt/kubernetes/commit/aeeaed6a7f750acd3afb900c35550ff4997ded30#diff-54fdd67617b0f2ef1b5a09148f77303eR1198

Reply:
are you suggesting that this pull should be rebasing after #79604?

Reply:
This PR can proceed first. It's fine for it to lift that method and export it (or place it in a package where that validation method could access it).

Reply:
copied and exported from validation pkg